### PR TITLE
Use ExceptionValueMap to store exception values (codes or message) instead of trait

### PR DESCRIPTION
**Pull request by munkie**
Use ExceptionValueMap to store map of exception values (codes or messages)
And use it instead of ClassMapHandlerTrait to get code or message associated with exception
In this case resolve logic is incapsulated in ExceptionValueMap and is not exposed into controller or serialize normalizer.

### CODE REVIEW

File: Resources/config/exception_listener.xml [hunk 1]
```diff
@@ -22,14 +22,22 @@
             <argument type="service" id="templating.engine.twig" />
         </service>
 
+        <service id="fos_rest.exception.codes_map" class="FOS\RestBundle\Util\ExceptionValueMap">
+            <argument type="collection"/> <!-- exception codes -->
+        </service>
+
+        <service id="fos_rest.exception.messages_map" class="FOS\RestBundle\Util\ExceptionValueMap">
```
Comment (Ener-Getick) [inline Resources/config/exception_listener.xml]: should be private (``public="false"``)

File: Util/ExceptionValueMap.php [hunk 2]
```diff
@@ -11,24 +11,35 @@
...
+    /**
      * Resolves the value corresponding to a class from an array.
      *
      * @param string $class
-     * @param array  $map
      *
      * @return mixed|false if not found
      */
-    public function resolveValue($class, array $map)
+    public function resolveClass($class)
```
Comment (Ener-Getick) [inline Util/ExceptionValueMap.php]: I don't think that's a good idea to have a function dedicated to the tests.

**Comment by Ener-Getick** [comment]
Seems really good :+1:

**Comment by Ener-Getick** [comment]
should be rebased on master

**Comment by lsmith77** [comment]
do you want to put this into 2.0?

**Comment by Ener-Getick** [comment]
Yeah it certainly don't affect people and it would be easier to respect bc

File: Util/ExceptionValueMap.php (hunk 2 above)
Comment (munkie) [inline Util/ExceptionValueMap.php]: So what's your suggestion: make it private ?

**Comment by munkie** [comment]
Rebased.
Made map services private.
Made ExceptionValueMap::resolveClass method private

**Comment by Ener-Getick** [comment]
Status: Reviewed

@lsmith77 what do you think about removing the required statuses ? It's really annoying to have to rebase a PR each another one is merged :-/

**Comment by lsmith77** [comment]
the conflicts is I think nothing related to the required checks .. this is more that git cannot cleanly merge the source branch into the target branch

**Comment by Ener-Getick** [comment]
@lsmith77 right here yeah but the other rebase were needed by github, nothing related to git. (and BTW it doesn't mean removing the protection against force pushing)

**Comment by Ener-Getick** [comment]
Merged in #1380 thank you @munkie

**Pull request closed without merging**